frontend/src/features/admin-form/create/workflow/components/WorkflowContent/EditStepBlock/RespondentBlock.tsx

```diff
@@ -170,19 +204,426 @@ const DynamicRespondentOption = ({
                 },
               }}
               render={({ field: { value = '', ...rest } }) => (
-                <>
+                <SingleSelect
+                  isDisabled={isLoading}
+                  isClearable={false}
+                  placeholder="Select a field"
+                  items={emailFieldItems}
+                  value={value}
+                  {...rest}
+                />
+              )}
+            />
+            <FormErrorMessage>{errors.field?.message}</FormErrorMessage>
+          </FormControl>
+        ) : null}
+      </Radio>
+    </>
+  )
+}
+
+interface ConditionalRoutingOptionProps extends RespondentOptionProps {
+  conditionalFormFields: FormFieldWithQuestionNo<
+    FormFieldDto<DropdownFieldBase>
+  >[]
+}
+
+export interface ConditionalRoutingConfig {
+  csvFile: File | null
+}
+
+const ConditionalRoutingOption = ({
```

Comment:
/nit might want to consider splitting this component file into smaller components file. We're roughly stablizing the feature set on this workflow (admin side of things), so we might not be revisiting this component again soon. 

```
EditStepBlock
├── RespondentBlock
  ├──Components
  │ ├── ConditionalRoutingOption.tsx
  │ ├── DynamicRespondentOption.tsx
  │ └── StaticRespondentOption.tsx
  ├── RespondentBlock.tsx
  └── index.tsx
```

Reply:
done in [ba0e126](https://github.com/opengovsg/FormSG/pull/7804/commits/ba0e126ba071e94e6ae7334b4a568081785288e3)